### PR TITLE
Add Discovery Host search and Provision support from API

### PR DESCRIPTION
**Pull request by jyejare**
Support is added:

- Discover host search

- Discover Host Provision

### CODE REVIEW

**Comment by ntkathole** [comment]
Can you please add api calls result and test coverage?

**Comment by jyejare** [comment]
@ntkathole The PR is still InProgress :) I will surely add that soon.

**Comment by jyejare** [comment]
For Test Results : https://github.com/SatelliteQE/robottelo/pull/7481#issue-340863395

**Comment by jyejare** [comment]
@ntkathole / @san7ket / @mirekdlugosz Can you ACK and  merge this PR ?

File: nailgun/entities.py [hunk 1]
```diff
@@ -1014,6 +1017,16 @@ def facts(self, synchronous=True, **kwargs):
         response = client.post(self.path('facts'), **kwargs)
         return _handle_response(response, self._server_config, synchronous)
 
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """Make sure, ``ip, mac`` are in the ignore list for read"""
```
Comment (ntkathole) [inline nailgun/entities.py]: nitpick : hostgroup and root_pass are also in ignore list

**Comment by jyejare** [comment]
Two Acks and hence merging.

**Pull request merged**